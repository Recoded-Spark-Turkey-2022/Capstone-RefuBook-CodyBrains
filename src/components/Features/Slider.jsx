import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import featuresData from '../../data/featuresData';
import star from '../../assets/star.svg';

// The Slider component is a pagination component that uses SwiperJS to display slides of content.
const Slider = () => {
  return (
    // The Swiper component is wrapped around the content that we want to display as slides.
    // We pass in the pagination prop and set clickable to true to enable pagination for the Swiper.
    // We also pass in the Pagination module to enable pagination functionality.
    <Swiper
      pagination={{
        clickable: true,
      }}
      modules={[Pagination]}
    >
      {/* We map over the featuresData array to render a SwiperSlide for each item in the array. */}
      {featuresData.map((item) => {
        return (
          // The SwiperSlide component is used to wrap the content that we want to display as a slide.
          // We pass a unique key prop to each SwiperSlide to help with rendering performance.
          <SwiperSlide key={item.id}>
            {/* The content for each slide is a div containing an image, a title, and some text. */}
            <div className=" flex flex-col items-center justify-center gap-5 py-5">
              <img src={star} alt="" />
              <p className="text-md  text-refubookLightBlack font-medium text-center">
                {item.title}
              </p>
              <p className="text-md  text-refubookLightBlack font-medium text-center mb-4">
                {item.text}
              </p>
            </div>
          </SwiperSlide>
        );
      })}
    </Swiper>
  );
};

export default Slider;
