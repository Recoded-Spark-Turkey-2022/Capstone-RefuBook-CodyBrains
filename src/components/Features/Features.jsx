import { Link } from 'react-router-dom';
import illustration from '../../assets/features-illustration.svg';
import Slider from './Slider';
import featuresData from '../../data/featuresData';
import Feature from './Feature';

// The Features component is a functional component that displays information about RefuBook.
const Features = () => {
  return (
    // The component consists of a div with a blue background and two main sections: a section with text and a button, and a section with an illustration and a list of features.
    <div className="w-full flex flex-col items-center justify-center bg-refubookAboutBlue py-4 mt-5">
      <div className=" w-3/4  md:w-2/3 flex flex-col items-center gap-4 justify-center py-5">
        {/* The first section contains a heading and a paragraph of text. */}
        <h1 className="text-4xl md:text-3xl xl:text-4xl   text-refubookBlue font-bold text-center">
          About
        </h1>
        <p className="text-md leading-relaxed md:text-lg  text-refubookLightBlack font-medium text-justify md:text-center">
          RefuBook is a website dedicated to helping refugees navigate their new
          surroundings and feel connected to their community. We offer a
          platform for sharing knowledge and experiences, and strive to be a
          comprehensive and reliable source of information. Thank you for
          visiting and we hope you find the resources here helpful in your
          journey as a refugee.
        </p>
        {/* The first section also contains a button with a link to the "about" page. */}
        <button
          type="button"
          className=" py-1 px-3 md:py-2 md:px-9 bg-refubookBlue text-refubookWhite font-bold text-lg md:text-xl  rounded-full shadow-lg hover:bg-refubookWhite hover:text-refubookBlack transition duration-500 ease-in-out"
        >
          <Link to="about">Learn More</Link>
        </button>
      </div>
      <div className="w-full flex flex-col items-center md:flex-row-reverse justify-center 2xl:justify-evenly gap-4 py-5">
        {/* The second section contains an illustration and a list of features. */}
        <img src={illustration} alt="" />
        {/* The Slider component is used to display the list of features on smaller screens. */}
        <div className="block md:hidden w-full max-w-sm">
          <Slider />
        </div>
        {/* On larger screens, the features are displayed in a grid using the Feature component. */}
        <div className="hidden md:grid grid-cols-2 grid-rows-2 gap-2 justify-start px-10">
          {/* We map over the featuresData array to render a Feature component for each item in the array. */}
          {featuresData.map((item) => {
            return <Feature key={item.id} item={item} />;
          })}
        </div>
      </div>
    </div>
  );
};

export default Features;
