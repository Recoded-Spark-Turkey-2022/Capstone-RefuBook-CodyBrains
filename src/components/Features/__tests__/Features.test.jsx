import React from 'react';
import { render } from '@testing-library/react';
/* This line imports the render function from the @testing-library/react library, which is used to render React */
import Features from '../Features';
import Feature from '../Feature';

/*

jest.mock('react-router-dom', () => ({ Link: 'Link', })); -
jest.mock('swiper/react', () => ({ Swiper: 'Swiper', SwiperSlide: 'SwiperSlide', })); - 
jest.mock('swiper', () => ({ Pagination: 'Pagination', })); - 
describe('Features', () => { ... }); - 
it('renders correctly', () => { ... }); - This block creates a test case using Jest's it function. The test case
const { container } = render(<Features />); - This line renders the Features component and destructures the container element from the returned object.
expect(container).toMatchSnapshot(); - ThisÃŸ line uses Jest's toMatchSnapshot function to create a snapshot of the rendered element and compare it to a previously saved snapshot. If the snapshots do not match, the test will fail.
it('renders Feature component with props passed', () => { ... }); - This block creates another test case using Jest's it function. The test case checks that the Feature component renders correctly when passed a mock item prop by taking a snapshot of the rendered element and comparing it to a previously saved snapshot.
const { feature } = render(<Feature key={1} item={{ title: 'Test Title', text: 'Test Text' }} />); - This line renders the Feature component and passes in a mock item prop. The feature element is destructured from the returned
*/

/*  This line mocks the react-router-dom library using the jest.mock function. 
This is necessary because the Features and Feature components are dependent on this library, but we don't want to actually include it in the test. 
Instead, we provide a fake implementation of the Link component, which allows us to run the test without running into errors. */
jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));

/* This line mocks the swiper/react library in the same way as the previous line. 
We provide fake implementations of the Swiper and SwiperSlide components, which allows us to run the test without running into errors. */
jest.mock('swiper/react', () => ({
  Swiper: 'Swiper',
  SwiperSlide: 'SwiperSlide',
}));

jest.mock('swiper', () => ({
  Pagination: 'Pagination',
}));

describe('Features', () => {
  /*  This block creates a test suite for the Features component using Jest's describe function. 
All of the test cases within this block will be run as part of the same test suite. */
  it('renders correctly', () => {
    /* checks that the Features component renders correctly by taking a snapshot of the rendered element and comparing it to a previously saved snapshot. */
    const { container } = render(<Features />);
    expect(container).toMatchSnapshot();
  });
  it('renders Feature component with props passed', () => {
    const { feature } = render(
      <Feature key={1} item={{ title: 'Test Title', text: 'Test Text' }} />
    );
    expect(feature).toMatchSnapshot();
  });
});
