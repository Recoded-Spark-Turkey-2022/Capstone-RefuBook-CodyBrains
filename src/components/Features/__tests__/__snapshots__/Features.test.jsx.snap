// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features renders Feature component with props passed 1`] = `undefined`;

exports[`Features renders correctly 1`] = `
<div>
  <div
    class="w-full flex flex-col items-center justify-center bg-refubookAboutBlue py-4 mt-5"
  >
    <div
      class=" w-3/4  md:w-2/3 flex flex-col items-center gap-4 justify-center py-5"
    >
      <h1
        class="text-4xl md:text-3xl xl:text-4xl   text-refubookBlue font-bold text-center"
      >
        About
      </h1>
      <p
        class="text-md leading-relaxed md:text-lg  text-refubookLightBlack font-medium text-justify md:text-center"
      >
        RefuBook is a website dedicated to helping refugees navigate their new surroundings and feel connected to their community. We offer a platform for sharing knowledge and experiences, and strive to be a comprehensive and reliable source of information. Thank you for visiting and we hope you find the resources here helpful in your journey as a refugee.
      </p>
      <link
        to="about"
      >
        <button
          class=" py-1 px-3 md:py-2 md:px-9 bg-refubookBlue text-refubookWhite font-bold text-lg md:text-xl  rounded-full shadow-lg hover:bg-refubookWhite hover:text-refubookBlack transition duration-500 ease-in-out"
          type="button"
        >
          Learn More
        </button>
      </link>
    </div>
    <div
      class="w-full flex flex-col items-center md:flex-row-reverse justify-center 2xl:justify-evenly gap-4 py-5"
    >
      <img
        alt=""
        src="features-illustration.svg"
      />
      <div
        class="block md:hidden w-full max-w-sm"
      >
        <swiper
          modules="Pagination"
          pagination="[object Object]"
        >
          <swiperslide>
            <div
              class=" flex flex-col items-center justify-center gap-5 py-5"
            >
              <img
                alt=""
                src="star.svg"
              />
              <p
                class="text-md  text-refubookLightBlack font-medium text-center"
              >
                Reliability
              </p>
              <p
                class="text-md  text-refubookLightBlack font-medium text-center mb-4"
              >
                At RefuBook, we prioritize accuracy and up-to-date information to be a dependable source for refugees. Trustworthiness is crucial for us.
              </p>
            </div>
          </swiperslide>
          <swiperslide>
            <div
              class=" flex flex-col items-center justify-center gap-5 py-5"
            >
              <img
                alt=""
                src="star.svg"
              />
              <p
                class="text-md  text-refubookLightBlack font-medium text-center"
              >
                Motivation
              </p>
              <p
                class="text-md  text-refubookLightBlack font-medium text-center mb-4"
              >
                Our goal is to help refugees feel connected to their community and succeed in their new home. We are dedicated to providing the necessary resources and support.
              </p>
            </div>
          </swiperslide>
          <swiperslide>
            <div
              class=" flex flex-col items-center justify-center gap-5 py-5"
            >
              <img
                alt=""
                src="star.svg"
              />
              <p
                class="text-md  text-refubookLightBlack font-medium text-center"
              >
                Efficiency
              </p>
              <p
                class="text-md  text-refubookLightBlack font-medium text-center mb-4"
              >
                We value efficiency and strive to make our website user-friendly for refugees to easily access the information and resources they need.
              </p>
            </div>
          </swiperslide>
          <swiperslide>
            <div
              class=" flex flex-col items-center justify-center gap-5 py-5"
            >
              <img
                alt=""
                src="star.svg"
              />
              <p
                class="text-md  text-refubookLightBlack font-medium text-center"
              >
                Creativity
              </p>
              <p
                class="text-md  text-refubookLightBlack font-medium text-center mb-4"
              >
                Creativity is essential for finding innovative solutions to the challenges refugees face. We encourage users to share their unique perspectives and ideas on our platform.
              </p>
            </div>
          </swiperslide>
        </swiper>
      </div>
      <div
        class="hidden md:grid grid-cols-2 grid-rows-2 gap-2 justify-start px-10"
      >
        <div
          class=" max-w-sm flex flex-col items-center   gap-4 py-5"
        >
          <div
            class="flex items-center justify-start w-full gap-4"
          >
            <img
              alt=""
              src="star.svg"
            />
            <h4
              class="text-lg text-refubookLightBlack text-left font-bold"
            >
              Reliability
            </h4>
          </div>
          <p
            class="text-md text-refubookLightBlack font-medium text-left"
          >
            At RefuBook, we prioritize accuracy and up-to-date information to be a dependable source for refugees. Trustworthiness is crucial for us.
          </p>
        </div>
        <div
          class=" max-w-sm flex flex-col items-center   gap-4 py-5"
        >
          <div
            class="flex items-center justify-start w-full gap-4"
          >
            <img
              alt=""
              src="star.svg"
            />
            <h4
              class="text-lg text-refubookLightBlack text-left font-bold"
            >
              Motivation
            </h4>
          </div>
          <p
            class="text-md text-refubookLightBlack font-medium text-left"
          >
            Our goal is to help refugees feel connected to their community and succeed in their new home. We are dedicated to providing the necessary resources and support.
          </p>
        </div>
        <div
          class=" max-w-sm flex flex-col items-center   gap-4 py-5"
        >
          <div
            class="flex items-center justify-start w-full gap-4"
          >
            <img
              alt=""
              src="star.svg"
            />
            <h4
              class="text-lg text-refubookLightBlack text-left font-bold"
            >
              Efficiency
            </h4>
          </div>
          <p
            class="text-md text-refubookLightBlack font-medium text-left"
          >
            We value efficiency and strive to make our website user-friendly for refugees to easily access the information and resources they need.
          </p>
        </div>
        <div
          class=" max-w-sm flex flex-col items-center   gap-4 py-5"
        >
          <div
            class="flex items-center justify-start w-full gap-4"
          >
            <img
              alt=""
              src="star.svg"
            />
            <h4
              class="text-lg text-refubookLightBlack text-left font-bold"
            >
              Creativity
            </h4>
          </div>
          <p
            class="text-md text-refubookLightBlack font-medium text-left"
          >
            Creativity is essential for finding innovative solutions to the challenges refugees face. We encourage users to share their unique perspectives and ideas on our platform.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
